### PR TITLE
[formrecognizer] Build model update

### PR DESCRIPTION
**Pull request by catalinaperalta**
Updating `begin_build_model()` to be more flexible for future design work. Also renaming `source` to `blob_container_url`.

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_document_model_administration_client.py [hunk 1]
```diff
@@ -93,27 +93,27 @@ def __init__(self, endpoint: str, credential: Union[AzureKeyCredential, TokenCre
 
     @distributed_trace
     def begin_build_model(
-        self, source: str, build_mode: Union[str, DocumentBuildMode], **kwargs: Any
+        self, build_mode: Union[str, DocumentBuildMode], blob_container_url: str, **kwargs: Any
```
Comment (catalinaperalta) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_document_model_administration_client.py:96]: This design would help with future content source overload flexibility.

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_document_model_administration_client.py [hunk 2]
```diff
@@ -93,27 +93,27 @@ def __init__(self, endpoint: str, credential: Union[AzureKeyCredential, TokenCre
...
         :param build_mode: The custom model build mode. Possible values include: "template", "neural".
             For more information about build modes, see: https://aka.ms/azsdk/formrecognizer/buildmode.
+        :param str blob_container_url: An Azure Storage blob container's SAS URI. A container URI (without SAS)
+            can be used if the container is public or has a managed identity configured. For more information on
+            setting up a training data set, see: https://aka.ms/azsdk/formrecognizer/buildtrainingset.
         :type build_mode: str or :class:`~azure.ai.formrecognizer.DocumentBuildMode`
         :keyword str model_id: A unique ID for your model. If not specified, a model ID will be created for you.
         :keyword str description: An optional description to add to the model.
-        :keyword str prefix: A case-sensitive prefix string to filter documents in the source path.
+        :keyword str prefix: A case-sensitive prefix string to filter documents in the blob_container_url path.
```
Comment (catalinaperalta) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_document_model_administration_client.py:116]: fix

**Review by kristapratico — APPROVED** [review]
LGTM

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,8 @@
 - Added `TargetAuthorization` of type `dict[str, str]`.
 
 ### Breaking Changes
+- Renamed `source` argument to `blob_container_url` on `begin_build_model()`.
+- Changed argument order on `begin_build_model()`. `build_mode` is the first expected argument, followed by `blob_container_url`.
```
Comment (samvaity) [inline sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:10]: curious why we made this change?

**Pull request merged**